src/FieldWarning/Assets/Units/Component/Movement/PathfinderData.cs

```diff
@@ -435,8 +436,10 @@ public void RemoveArc(PathNode node1, PathNode node2)
         /// <summary>
         /// Run the A* algorithm and put the result in path.
         /// If no path was found, return 'forever' and put only the destination in path.
+        /// Note: this method MUST be syncrhonized as it is called from multiple threads.
         /// Returns the total path time.
-        /// </summary>
+        /// </summary> 
+        [MethodImpl(MethodImplOptions.Synchronized)]
```

Comment:
Since this just resolves to a lock, I would find it more readable if you take the lock manually in the method. But maybe a real C# programmer would not need to google the annotations as I had to. So, up to you.

Reply:
will remove that and use events instead